lib/vmpooler/api/v1.rb

```diff
@@ -173,6 +173,72 @@ class V1 < Sinatra::Base
       JSON.pretty_generate(result)
     end
 
+    get "#{api_prefix}/token/:token/?" do
+      content_type :json
+
+      result = {}
+
+      Vmpooler::API.settings.config[:auth] ? status(401) : status(404)
+      result['ok'] = false
+
+      if Vmpooler::API.settings.config[:auth] and Vmpooler::API.settings.redis.exists('vmpooler__token__' + params[:token])
+        status(200)
+        result['ok'] = true
+
+        result[params[:token]] = Vmpooler::API.settings.redis.hgetall('vmpooler__token__' + params[:token])        
```

Comment:
Can the `exists` call be skipped? This would change the `404` assumption to mean does not exist **or** has no content.

``` ruby
if config[:auth]
  token_details = redis.hgetall(...)
end

unless token_details.nil?
  status(200)
  result['ok'] = true
end
```